gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,260 @@
+            v2_subpop=pop_ht.project_meta.v2_subpop,
+            pop=pop_ht.population_inference.pop,
+            project_id=pop_ht.project_meta.project_id,
+            project_pop=pop_ht.project_meta.project_pop,
+        )
+        pop_ht = pop_ht.checkpoint(ht_path, overwrite=args.overwrite)
+    else:
+        pop_ht = hl.read_table(ht_path)
+
+    return pop_ht
+
+
+def main(args):
+
+    # Read in the raw mt
+    mt = get_gnomad_v3_mt(key_by_locus_and_alleles=True)
+
+    # Filter to test partitions if specified
+    if args.test:
+        logger.info("Filtering to the first two partitions of the MT")
+        mt = mt._filter_partitions(range(2))
+
+    # Write out the densified MT
+    output_path = (
+        get_sample_qc_root()
+        + "/subpop_analysis/"
+        + f"gnomad_v{CURRENT_VERSION}_qc_mt_subpop_analysis.mt"
+    )
+
+    if args.make_full_subpop_qc_mt:
+        logger.info("Generating densified MT to use for all subpop analyses...")
+        mt = compute_subpop_qc_mt(mt, args.min_popmax_af)
+        mt = mt.naive_coalesce(5000)
+        mt = mt.checkpoint(output_path, overwrite=args.overwrite)
+
+    logger.info("Generating PCs for subpops...")
+    # TODO: Add code for AMR using currently unused `run_pop_pca` function
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="This script generates a QC MT and PCA scores to use for subpop analyses"
+    )
+    parser.add_argument(
+        "--min-popmax-af",
+        help="Minimum population max variant allele frequency to retain a variant for the subpop QC MT",
+        type=float,
+        default=0.001,
+    )
+    parser.add_argument(
+        "--pop",
+        help="Population to which the subpop QC MT should be filtered when generating the PCA data",
+        type=str,
+    )
+    parser.add_argument(
+        "--min-af",
+        help="Minimum population variant allele frequency to retain variant in QC MT when generating the PCA data",
+        type=float,
+        default=0.001,
+    )
+    parser.add_argument(
+        "--min-inbreeding-coeff-threshold",
+        help="Minimum site inbreeding coefficient to keep a variant when generating the PCA data",
+        type=float,
+        default=-0.25,
+    )
+    parser.add_argument(
+        "--outlier-ht-path",
+        help="Path to Table keyed by column containing outliers to remove when generating the PCA data",
+        type=str,
+    )
+    parser.add_argument(
+        "--outlier-string",
+        help="String used to describe which outliers(if any were set by 'outlier-ht-path') were removed when generating the PCA data",
+        type=str,
+        default="",
+    )
+    parser.add_argument(
+        "--make-full-subpop-qc-mt",
+        help="Runs function to create dense MT to use as QC MT for all subpop analyses. This uses --min-popmax-af to determine variants that need to be retained",
+        action="store_true",
+    )
```

Comment:
Rearrange arguments a bit, and add Slack option. I would put slack, overwrite, and test at the beginning then ---make-full-subpop-qc-mt, then --min-popmax-af, then make a parameter that will be used for the PCA, and at this point I think you can just add in the call to that function, you have already done all the legwork at this point